server/src/test/java/org/elasticsearch/index/mapper/RoutingPathFieldsTests.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.index.mapper;
+
+import org.elasticsearch.cluster.metadata.IndexMetadata;
+import org.elasticsearch.cluster.routing.IndexRouting;
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.index.IndexSettings;
+import org.elasticsearch.index.IndexVersion;
+import org.elasticsearch.test.ESTestCase;
+
+import java.util.Arrays;
+
+public class RoutingPathFieldsTests extends ESTestCase {
```

Comment:
The tests below are pretty basic.. There's coverage too in `IdLoaderTests` and `TimeSeriesIdFieldMapperTests`, but suggestions are welcome on how to improve this one.

Reply:
Maybe add a test that ensures that hashing the same content multiple times produces the same hash.
And maybe add a unit test for `decodeAsMap(...)` method?

Reply:
Added stability checks. `decodeAsMap` is tricky to test as it requires a properly formatted input, it's implicitly covered in `TimeSeriesIdFieldMapperTests`